### ISSUE TITLE
Exclude package manager lock files

### ISSUE THREAD

**Issue by JoostVoskuil** [zricethezav/gitleaks#1098]
**Describe the bug**
False positives on pnpm-lock.yaml files that is used within NPM. The file contains a package called 'token'. This will lead to false positive 'generic-api-key: Generic API Key' 

**To Reproduce**
Create a file with the following data: 

  /@material/slider/15.0.0-canary.684e33d25.0:
    resolution: {integrity: sha512-WVyK+2pSNSZmj07M2K/a3TADoQ9FBCndfNC/vE7/wGIg4dddJJK5KvQ+yruf9R2cSzTL/S1sZ5WpyyeM8E9HTw==}
    dependencies:
      **'@material/tokens': 15.0.0-canary.684e33d25.0**
      '@material/typography': 15.0.0-canary.684e33d25.0
      tslib: 2.5.0
    dev: false

**Expected behavior**
No false positives

**Additional context**
Shoudn't package-lock.json and pnpm-lock.yaml be whitelisted completely? These are auto-generated files created by package managers (in this case npm and pnpm)

Two questions:
1) Do we want to whitelist package-lock.json and pnpm-lock.yaml completely?
2) Where do we want to exclude them? In the generic.go file? I need some help / an example how to exclude files. I have absolutely no idea how you can exclude files. Tried to read the docs but I am new in the go language. Can someone help me? This does not work ;)

``` generic.go
Allowlist: config.Allowlist{
	StopWords: DefaultStopWords,
	Paths: []*regexp.Regexp{
		regexp.Match("^pnpm-lock\\.yaml$"),
		regexp.Match("^package-lock\\.yaml$"),
	},
},
```

cc @zricethezav

**Comment by JoostVoskuil** [comment]
I see that https://github.com/zricethezav/gitleaks/pull/1076 partly covers this issue. But the toml file is autogenerated so including the files by directly editing the toml file is not the way to go right?

**Comment by lawndoc** [comment]
> the toml file is autogenerated so including the files by directly editing the toml file is not the way to go right?

They already pointed that out and had me updated the config template instead. Still waiting on their feedback after I made that change.

**Comment by zricethezav** [comment]
@JoostVoskuil @lawndoc merged #1076
